javalib/src/main/scala/java/util/concurrent/Semaphore.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.concurrent
+
+import java.util.{Collection, Collections}
+
+class Semaphore(var permits: Int, fairness: Boolean) extends Serializable {
+
+  def this(permits: Int) = this(permits, false)
+
+  // These methods can't be implemented because calling it more times than the number
+  // of permits makes no sense in Javascript.
+  // def acquire(): Unit = ???
+  // def acquire(permits: Int): Unit = ???
+  // def acquireUninterruptibly(): Unit = this.acquire()
+  // def acquireUninterruptibly(permits: Int): Unit = this.acquire(permits)
+
+  def availablePermits(): Int = this.permits
```

Comment:
We usually do not qualify `this` if it is not necessary.

Reply:
:+1: